File: C++/프린터.cpp
#include <string>
#include <vector>
#include <deque>
#include <queue>
#define X first
#define Y second
using namespace std;
priority_queue<int> pq;
deque<pair<int, int>> dq;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    for(int i=0; i<priorities.size(); i++) {
        dq.push_back({priorities[i], i});
        pq.push(priorities[i]);
    }
    while(!dq.empty()) {
        auto curr = dq.front();
        dq.pop_front();
        int top = pq.top();
        if(curr.X == top) {
            answer++;
            pq.pop();
            if(curr.Y == location) break;
        }
        else dq.push_back(curr);
    }
    return answer;
}

File: python/Programmers/프린터.py
from collections import deque
def solution(pri,location):
    answer = 0
    pri = deque(pri)
    result = deque(([0]*len(pri)))
    result[location] = 1
    while pri:
        print(pri)
        print(result)
        if pri[0] == max(pri):
            pri.popleft()
            if result.popleft() == 1:
                answer += 1
                break
            else:
                answer += 1
        else:
            pri.append(pri.popleft())
            result.append(result.popleft())
    return answer
print(solution([1, 1, 9, 1, 1, 1],0))
